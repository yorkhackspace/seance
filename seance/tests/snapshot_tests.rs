#[test]
fn hackspace_logo() {
    let design_file = seance::svg::parse_svg(
        &std::fs::read("../logo.svg").expect("failed to read logo SVG file"),
    )
    .expect("failed to parse logo SVG data");

    let design_name = "York Hackspace Logo";
    let offset = seance::DesignOffset { x: 0.0, y: 0.0 };

    let mut tool_passes = seance::default_passes::default_passes();
    for tool in &mut tool_passes {
        tool.set_enabled(true);
    }

    let bed = &seance::bed::BED_GCC_SPIRIT;

    let paths = seance::svg::get_paths_grouped_by_colour(&design_file);
    insta::assert_debug_snapshot!("york hackspace logo SVG paths", &paths);

    let mut paths_in_mm = seance::resolve_paths(&paths, &offset, 1.0);
    insta::assert_debug_snapshot!("york hackspace logo resolved paths", &paths_in_mm);

    seance::paths::filter_paths_to_tool_passes(&mut paths_in_mm, &tool_passes);
    insta::assert_debug_snapshot!("york hackspace logo filtered paths", &paths_in_mm);

    let resolved_paths = seance::paths::convert_points_to_plotter_units(&paths_in_mm, &bed);
    insta::assert_debug_snapshot!("york hackspace logo plotted paths", &resolved_paths);

    let hpgl = seance::hpgl::generate_hpgl(&resolved_paths, &tool_passes, &bed);
    insta::assert_snapshot!("york hackspace logo HPGL output", &hpgl);

    let pcl = seance::pcl::wrap_hpgl_in_pcl(hpgl, design_name, &tool_passes);
    insta::assert_snapshot!("york hackspace logo PCL output", &pcl);
}
