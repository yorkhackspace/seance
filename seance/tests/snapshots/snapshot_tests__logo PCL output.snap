---
source: seance/tests/snapshot_tests.rs
expression: "&pcl"
---
%-12345XE!m4NLogo!v16R1111111111111111!v64I0400040004000400040004000400040004000400040004000400040004000400!v64V0020002000200020002000200020002000200020002000200020002000200020!v64P0100010001000100010001000100010001000100010001000100010001000100!v16D*t508R&u508R!r0N%1A!r1000I!r1000K!r500P*t508R&u508R!m0S!s1S%1BIN;SC;PU;SP1;LT;PU0,18528;PU0,18528;SP0;EC0;EC1;OE;%1AE%-12345X
